### PR TITLE
feat: removes provider from init and moves CI templates

### LINKED ISSUE #343: Move workflow templates out of trestlebot package
**Is your feature request related to a problem? Please describe.**

Including the GitHub workflows within the `trestle-bot` package will result in outdated workflows. This will lead to issues with a seamless initialization.

**Describe the solution you'd like**

- Move the workflow from the package to a top-level `TEMPLATES` directory and let the user copy the most up to date workflow manually. 
- Remove provider from `init`

**Describe alternatives you've considered**
- Having init pull them down over HTTPS, but this could present challenges in airgapped workflows

**Additional context**
Will result in the closure of #335

### PR DESCRIPTION
**Pull request by jpower432**
## Description

This moves the CI templates out from the `trestle-bot` packages and allows users to copy them from `TEMPLATES`
Fixes #343 

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested?

- [x] Test each workflow file - https://github.com/jpower432/trestlebot-demo-env/actions

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gvauter — APPROVED** [review]
LGTM

File: docs/tutorials/github.md [hunk 1]
```diff
@@ -77,6 +74,14 @@ Our `profile.json` file contains a reference to our `catalog.json` file.  By def
 sed -i 's/NIST_SP-800-53_rev5_catalog.json/trestle:\/\/catalogs\/nist_rev5_800_53\/catalog.json/g' profiles/nist_rev5_800_53/profile.json
 ```
 
+Finally you can copy ready-made CI/CD workflows from the `TEMPLATES` directory into your workspace. These are the trestlebot actions that will run as we make changes to the repo contents.
+
+**For example Component Definition authoring in GitHub Actions**
+```
+cp TEMPLATES/github/trestlebot-create-component-definition.yml .github/workflows
```
Comment (huiwangredhat) [inline docs/tutorials/github.md:81]: The _.github_ dir won't be generated after the trestlebot init and has been removed on line 53. So _.github_ needs to be created. Is that the expected?

**Comment by huiwangredhat** [comment]
The job failed after copying the workflow .github/workflows/trestlebot-rules-transform.yml. More details: https://github.com/huiwangredhat/trestlebot-demo3/actions/runs/10842292114. This file needs to be updated by the end users according to the requirements, am I right?

File: docs/tutorials/github.md (hunk 1 above)
Comment (jpower432) [reply docs/tutorials/github.md:81]: Yes. The `.github/workflows` directory would need to be created. Thanks for pointing that out. I could include a `mkdir` line.

**Comment by jpower432** [comment]
> The job failed after copying the workflow .github/workflows/trestlebot-rules-transform.yml. More details: https://github.com/huiwangredhat/trestlebot-demo3/actions/runs/10842292114(I guess it should be _contents: write_). The workflow files are allowed to be updated by the end users according to the requirements, am I right?

You are right! Thank you finding this issue.

**Review by huiwangredhat — APPROVED** [review]
Thanks for your reply. LGTM.

**Pull request merged**